tests/integration/ha_tests/test_restore_cluster.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+import logging
+
+import pytest
+from pytest_operator.plugin import OpsTest
+
+from tests.integration.ha_tests.helpers import (
+    METADATA,
+    add_unit_with_storage,
+    app_name,
+    get_patroni_cluster,
+    reused_storage,
+    storage_id,
+    storage_type,
+)
+from tests.integration.helpers import (
+    CHARM_SERIES,
+    db_connect,
+    get_password,
+    get_primary,
+    get_unit_address,
+)
+
+APP_NAME = METADATA["name"]
+SECOND_APPLICATION = "second-cluster"
+
+logger = logging.getLogger(__name__)
+
+charm = None
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest) -> None:
+    """Build and deploy three unit of PostgreSQL."""
+    wait_for_apps = False
+    # It is possible for users to provide their own cluster for HA testing. Hence, check if there
+    # is a pre-existing cluster.
```

Comment:
This is a potentially destructive test, does this assumption hold true here as well?

Reply:
let's do not allow running this test with existing installations.